### ISSUE TITLE
Improve Screen Reader on Heading block.

### ISSUE THREAD

**Issue by etoledom** [wordpress-mobile/gutenberg-mobile#908]
Unselected state: 
- Read block type.
- Read content. (Maybe heading level too?)
- Read "Empty" if there is no content.

Selected State: 
- The block should be set as selected when it's focused by the Screen Reader.
- Access RichText to edit.
- Access Toolbar Buttons.
- Hide Keyboard button should work as expected.

Links UI depends on: https://github.com/wordpress-mobile/gutenberg-mobile/issues/914
